eradiate/scenes/biosphere/_discrete.py

```diff
@@ -909,6 +1028,159 @@ def bsdfs(self, ctx=None):
             }
         }
 
+
+@CanopyElementFactory.register("abstract_tree")
+@parse_docs
+@attr.s
+class AbstractTree(CanopyElement):
+    leaf_cloud = documented(
+        attr.ib(
+            default=None,
+            converter=attr.converters.optional(LeafCloud.convert),
+            validator=attr.validators.optional(attr.validators.instance_of(LeafCloud)),
+        ),
+        doc="Instanced leaf cloud. Can be specified as a dictionary, which will "
+        "be interpreted by :meth:`.LeafCloud.from_dict`.",
+        type=":class:`LeafCloud`",
+        default="None",
+    )
+
+    trunk_height = documented(
+        pinttr.ib(default=0.0 * ureg.m, units=ucc.deferred("length")),
+        doc="Trunk height. If unset defaults to zero, yielding no trunk.\n\n"
```

Comment:
I think the "no trunk" case is irrelevant since you already have `LeafCloud` for this.